### PR TITLE
Add laundry endpoint + logging

### PR DESCRIPTION
**Pull request by shinehlina**

### CODE REVIEW

File: laundry-management-service/app/src/main/java/com/microservices/laundrymanagement/controllers/LaundryController.java [hunk 1]
```diff
@@ -1,27 +1,43 @@
...
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.web.bind.annotation.*;
 
 @RestController
 @RequestMapping("laundry")
 public class LaundryController {
     private OrderService orderService;
+    private LaundryService laundryService;
+    private final Logger logger = LoggerFactory.getLogger(LaundryController.class);
```
Comment (andrsuh) [inline laundry-management-service/app/src/main/java/com/microservices/laundrymanagement/controllers/LaundryController.java:13]: private final have precedence over private fields/ move them higher

File: laundry-management-service/app/src/main/java/com/microservices/laundrymanagement/controllers/LaundryController.java [hunk 2]
```diff
@@ -1,27 +1,43 @@
...
 import com.microservices.laundrymanagement.service.OrderService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.web.bind.annotation.*;
 
 @RestController
 @RequestMapping("laundry")
 public class LaundryController {
```
Comment (andrsuh) [inline laundry-management-service/app/src/main/java/com/microservices/laundrymanagement/controllers/LaundryController.java:12]: order related stuff should be moved into order controller I suppose

File: laundry-management-service/app/src/main/java/com/microservices/laundrymanagement/service/LaundryService.java [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package com.microservices.laundrymanagement.service;
+
+public interface LaundryService {
+    int addNewLaundry();
```
Comment (andrsuh) [inline laundry-management-service/app/src/main/java/com/microservices/laundrymanagement/service/LaundryService.java]: maybe "registerLaundry" is better?

File: laundry-management-service/app/src/main/java/com/microservices/laundrymanagement/service/impl/LaundryServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class LaundryServiceImpl implements LaundryService {
+    private LaundryStateRepository laundryStateRepository;
+
+    private final Logger logger = LoggerFactory.getLogger(LaundryServiceImpl.class);
```
Comment (andrsuh) [inline laundry-management-service/app/src/main/java/com/microservices/laundrymanagement/service/impl/LaundryServiceImpl.java:14]: field ordering

File: laundry-management-service/app/src/main/java/com/microservices/laundrymanagement/service/impl/LaundryServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+    @Autowired
+    public LaundryServiceImpl(LaundryStateRepository laundryStateRepository) {
+        this.laundryStateRepository = laundryStateRepository;
+    }
+
+    @Override
+    public int addNewLaundry() {
+        logger.info("Creating queue...");
```
Comment (andrsuh) [inline laundry-management-service/app/src/main/java/com/microservices/laundrymanagement/service/impl/LaundryServiceImpl.java]: not informative

File: laundry-management-service/app/src/main/java/com/microservices/laundrymanagement/controllers/LaundryController.java [hunk 6]
```diff
@@ -1,27 +1,43 @@
...
     @PutMapping
     public void completeOrder(@RequestParam int id) {
+        logger.info("Got completion request for order with id: {}", id);
         orderService.completeOrder(id);
     }
-}
+
+    @PostMapping("/new")
+    @ResponseStatus(HttpStatus.CREATED)
+    public int addLaundry() {
```
Comment (andrsuh) [inline laundry-management-service/app/src/main/java/com/microservices/laundrymanagement/controllers/LaundryController.java]: according to the tech spec this call should receive laundry name

File: laundry-management-service/app/src/main/java/com/microservices/laundrymanagement/service/impl/OrderServiceImpl.java [hunk 7]
```diff
@@ -35,6 +39,7 @@ public OrderServiceImpl(OrderRepository orderRepository,
     @Transactional
     @Override
     public void submitOrder(OrderSubmissionDto orderSubmissionDto) {
+        logger.info("Submitting order: {}...", orderSubmissionDto);
         OrderEntity orderEntity = new OrderEntity(orderSubmissionDto);
         if (orderRepository.existsById(orderSubmissionDto.getOrderId()))
```
Comment (andrsuh) [inline laundry-management-service/app/src/main/java/com/microservices/laundrymanagement/service/impl/OrderServiceImpl.java]: bad formatting

File: laundry-management-service/app/src/main/java/com/microservices/laundrymanagement/service/impl/OrderServiceImpl.java [hunk 8]
```diff
@@ -46,14 +51,17 @@ public void submitOrder(OrderSubmissionDto orderSubmissionDto) {
...
     }
 
     @Transactional
     @Override
     public void completeOrder(int id) {
+        logger.info("Completing order {}...", id);
         OrderEntity order = orderRepository.findById(id)
                 .orElseThrow(() -> new IllegalArgumentException("Order with id " + id + " is not found")); //todo: logs
         if (order.getStatus() == OrderStatus.COMPLETE) {
+            logger.warn("Order with id {} is already completed");
```
Comment (andrsuh) [inline laundry-management-service/app/src/main/java/com/microservices/laundrymanagement/service/impl/OrderServiceImpl.java]: In previous example you did not log this case

File: laundry-management-service/app/src/main/java/com/microservices/laundrymanagement/util/LaundryWorkingActivityEmulator.java [hunk 9]
```diff
@@ -19,6 +21,8 @@
 
     private final OrderService orderService;
 
+    private final Logger logger = LoggerFactory.getLogger(LaundryWorkingActivityEmulator.class);
```
Comment (andrsuh) [inline laundry-management-service/app/src/main/java/com/microservices/laundrymanagement/util/LaundryWorkingActivityEmulator.java:24]: ordering

**Review by andrsuh — CHANGES_REQUESTED** [review]

File: laundry-management-service/app/src/main/java/com/microservices/laundrymanagement/controllers/LaundryController.java (hunk 1 above)
Comment (shinehlina) [reply laundry-management-service/app/src/main/java/com/microservices/laundrymanagement/controllers/LaundryController.java:13]: Hmmm... I used to think that Idea deals with such things:)

**Review by andrsuh — APPROVED** [review]

**Pull request merged**